src/compiler/checker.ts

```diff
@@ -42870,6 +42870,9 @@ namespace ts {
         }
 
         function checkGrammarClassLikeDeclaration(node: ClassLikeDeclaration): boolean {
+            if (languageVersion >= ScriptTarget.ES2015 && node.parent && isIfStatement(node.parent)) {

```

Comment:
I don't think we want to gate this on the target.

Reply:
Can you provide some more guidance? :)

Reply:
Do you mean I should simply check always?